test/NuGet.Core.Tests/NuGet.Commands.Test/UnexpectedDependencyMessagesTests.cs

```diff
@@ -0,0 +1,592 @@
+
+            logs.Select(e => e.Code).ShouldAllBeEquivalentTo(NuGetLogCode.NU2504);
+            logs.Select(e => e.Level).ShouldAllBeEquivalentTo(LogLevel.Warning);
+            logs.Select(e => e.Message)
+                .ShouldBeEquivalentTo(new[]
+                {
+                    "Project dependency x (< 2.0.0) does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results.",
+                    "Project dependency y (< 3.0.0) does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results."
+                });
+        }
+
+        [Fact]
+        public void GivenAProjectWithMultipleFrameworksAndDifferentRangesVerifyDifferentWarningsPerPackage()
+        {
+            var tfi = new List<TargetFrameworkInformation>();
+            tfi.AddRange(GetTFI(NuGetFramework.Parse("net46"), new LibraryRange("x", VersionRange.Parse("(, 2.0.0)"), LibraryDependencyTarget.Package)));
+            tfi.AddRange(GetTFI(NuGetFramework.Parse("netstandard1.3"), new LibraryRange("x", VersionRange.Parse("(, 2.0.0]"), LibraryDependencyTarget.Package)));
+
+            var project = new PackageSpec(tfi)
+            {
+                Name = "proj"
+            };
+
+            UnexpectedDependencyMessages.GetProjectDependenciesMissingLowerBounds(project).Count().Should().Be(2);
+        }
+
+        [Fact]
+        public void GivenAProjectWithMultipleFrameworksAndSameDependenciesVerifyASingleWarningPerPackage()
+        {
+            var range = VersionRange.Parse("(, 2.0.0)");
+            var tfi = new List<TargetFrameworkInformation>();
+            tfi.AddRange(GetTFI(NuGetFramework.Parse("net46"), new LibraryRange("x", range, LibraryDependencyTarget.Package)));
+            tfi.AddRange(GetTFI(NuGetFramework.Parse("netstandard1.3"), new LibraryRange("x", range, LibraryDependencyTarget.Package)));
+
+            var project = new PackageSpec(tfi)
+            {
+                Name = "proj"
+            };
+
+            UnexpectedDependencyMessages.GetProjectDependenciesMissingLowerBounds(project).Count().Should().Be(1);
+        }
+
+        [Fact]
+        public void GivenAProjectWithADependencyOnAPackageWithANullRanageVerifyWarningMessage()
+        {
+            var tfi = GetTFI(NuGetFramework.Parse("net46"), new LibraryRange("x", null, LibraryDependencyTarget.Package));
+            var project = new PackageSpec(tfi)
+            {
+                Name = "proj"
+            };
+
+            var log = UnexpectedDependencyMessages.GetProjectDependenciesMissingLowerBounds(project).Single();
+
+            log.Code.Should().Be(NuGetLogCode.NU2504);
+            log.Message.Should().Be("Project dependency x does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results.");
```

Comment:
I feel that `version range == null` should be a different error than missing lower bound.